### ISSUE TITLE
Possible Issue with TF 0.5.0 AWS Security Groups

### ISSUE THREAD

**Issue by stack72** [hashicorp/terraform#1872]
I have the following resource:

```
resource "aws_security_group" "default" {
  name = "my-default-sg"
  description = "Default security group that allows inbound and outbound traffic from all instances in the VPC"
  vpc_id = "${aws_vpc.default.id}"

  ingress {
    from_port   = "0"
    to_port     = "0"
    protocol    = "-1"
    self        = true
  }

  tags {
    Name = "my-default-sg"
  }
}
```

In TF 0.4.2, this creates both inbound and outbound All Traffic rules. But on 0.5.0 it seems to only creates inbound rules

TF 0.4.2
![screen shot 2015-05-08 at 18 30 19](https://cloud.githubusercontent.com/assets/227823/7541873/56962516-f5b0-11e4-8490-064a11cda010.png)

TF 0.5.0
![screen shot 2015-05-08 at 18 32 31](https://cloud.githubusercontent.com/assets/227823/7541901/9d1e2ccc-f5b0-11e4-905e-f5207a69a0cc.png)

No code changes between these commits. This is just the matter of installing a different version of TF

**Comment by mitchellh** [comment]
This was a change with TF 0.5.0 we perhaps need to document better in the CHANGELOG. TF 0.5.0 actually "fixes" a bug with TF 0.4.2. With 0.4.x it'd create both on the _first_ apply, but then delete the outbound on the second apply (beacuse its not in the config). We now reconcile this all at once. 

The solution is to add the egress to your resource.

**Comment by stack72** [comment]
Thanks @mitchellh - that immediately fixed it :)
